jraft-core/src/main/java/com/alipay/sofa/jraft/core/NodeImpl.java

```diff
@@ -583,8 +587,38 @@ private void handleElectionTimeout() {
             }
             resetLeaderId(PeerId.emptyPeer(), new Status(RaftError.ERAFTTIMEDOUT, "Lost connection from leader %s.",
                 this.leaderId));
-            doUnlock = false;
-            preVote();
+
+            // priority 0 is a special value so that a node with zero priority will never be a leader.
+            if (this.serverId.getPriority() == 0) {
+                return;
+            }
+
+            // if current node's priority < target_priority, it does not initiate leader election
+            // and waits for the next election timeout
+            if (this.serverId.getPriority() != -1 && this.serverId.getPriority() < this.targetPriority) {
+                this.electionTimeOutCounter++;
+
+                // if next leader is not elected until next election timeout, it exponentially
+                // lessens its local target priority
+                if (this.electionTimeOutCounter == 2) {
+
+                    this.targetPriority = (int) (this.targetPriority * options.getLessenPriorityRatio());
+
+                    if (this.targetPriority == 0) {
+                        this.targetPriority = 1;
+                    }
```

Comment:
max(1, (int) (this.targetPriority * options.getLessenPriorityRatio())) ?

Reply:
Okay,I will optimize this codes.